spec/app/ics-interchain-queries/README.md

```diff
@@ -0,0 +1,325 @@
+
+- **Live chains:** Both the Querying and Queried chains MUST be live, i.e., new blocks are eventually added to the chain.
+
+- **Censorship-resistant Querying Chain:**  The Querying Chain cannot selectively omit transactions.
+
+- **Correct relayer:** There is at least one correct relayer between the Querying and Queried chains. This is required for liveness. 
+
+### Desired Properties
+
+#### Permissionless
+
+A Querying Chain can query a chain and implement cross-chain querying without any approval from a third party or chain governance. Note that since there is no prior negotiation between chains, the Querying Chain cannot assume that queried data will be in an expected format.
+
+#### Minimal Queried Chain Work
+
+A Queried Chain has to do no implementation work or add any module to enable cross-chain querying. By utilizing an RPC client on a relayer, this is possible.
+
+#### Modular
+
+Adding cross-chain querying should be as easy as implementing a module in your chain.
+
+#### Control Queried Data
+
+The Querying Chain should have ultimate control over how to handle queried data. Like querying for a certain query form/type.
+
+#### Incentivization
+
+A bounty is paid to incentivize relayers for participating in interchain queries.
+
+## Technical Specification
+
+### General Design 
+
+The Querying Chain MUST implement the Cross-chain Querying module, which allows the Querying Chain to query state at the Queried Chain. 
+
+Cross-chain querying relies on relayers operating between both chains. When a query request is received by the Querying Chain, the Cross-chain Querying module emits a `sendQuery` event. Relayers operating between the Querying and Queried chains must monitor the Querying chain for `sendQuery` events. Eventually, a relayer will retrieve the query request and execute it at the Queried Chain. The relayer then submits (on-chain) the result at the Querying Chain. The result is finally registered at the Querying Chain by the Cross-chain Querying module.
+
+A query request includes the height of the Queried Chain at which the query must be executed. The reason is that the keys being queried can have different values at different heights. Thus, a malicious relayer could choose to query a height that has a value that benefits it somehow. By letting the Querying Chain decide the height at which the query is executed, we can prevent relayers from affecting the result data.
+
+### Data Structures
+
+The Cross-chain Querying module stores query requests when it processes them. 
+
+A CrossChainQuery is a particular interface to represent query requests. A request is retrieved when its result is submitted.
+
+```typescript
+interface CrossChainQuery struct {
+    id: Identifier
+    path: CommitmentPath
+    timeoutHeight: Height
+    queryHeight: Height
+    clientId: Identifier
+    bounty: sdk.Coin
```

Comment:
I think it is a good idea. It is kind of weird to have cross-apps dependencies, isn't? We should find a way to define the fee interface somewhere else in the core specs. Since this is a relayer thingy, maybe the relayer should define it.

Reply:
The spec should not be dependent on SDK types